src/main/java/org/apache/commons/collections4/trie/analyzer/StringKeyAnalyzer.java

```diff
@@ -82,6 +82,9 @@ public int bitIndex(final String key, final int offsetInBits, final int lengthIn
                 k = 0;
             } else {
                 k = key.charAt(index1);
+                if (k == 0) {
+                    throw new IllegalArgumentException("Don't support '\\u0000' in the key.");
+                }
```

Comment:
A better message might be "Character '\\u0000' is not supported in the key."

Reply:
'\\u0000' is shown as \u0000, '\u0000' is shown as blank space.

Reply:
Could/Should this method be rewritten to allow char 0 instead?